tests/integration/steps/integration_steps.go

```diff
@@ -0,0 +1,622 @@
+		_, err = helm.RunHelmCommandAndGetOutputE(t, options, "install", "traefik", "traefik/traefik", "-n", "traefik", "--create-namespace", "-f", valuesPath)
+		g(t).Expect(err).To(gomega.BeNil())
+	}
+}
+
+type credentials struct {
+	username string
+	password string
+}
+
+func CheckKeyspaceExists(t *testing.T, keyspace string) {
+	keyspaces := runCassandraQueryAndGetOutput(t, "describe keyspaces")
+
+	// Check that the keyspace exists in the list of keyspaces
+	g(t).Expect(keyspaces).Should(gomega.ContainSubstring(keyspace))
+}
+
+func WaitForReaperPod(t *testing.T) {
+	WaitForPodWithLabelToBeReady(t, "app.kubernetes.io/managed-by=reaper-operator")
+}
+
+func CheckRowCountInTable(t *testing.T, nbRows int, tableName, keyspaceName string) {
+	output := runCassandraQueryAndGetOutput(t, fmt.Sprintf("SELECT id FROM %s.%s", keyspaceName, tableName))
+
+	// Check that we have the right number of rows
+	g(t).Expect(output).Should(gomega.ContainSubstring(fmt.Sprintf("(%d rows)", nbRows)))
+}
+
+func CreateCassandraTable(t *testing.T, tableName, keyspaceName string) {
+	runCassandraQueryAndGetOutput(t, fmt.Sprintf("CREATE KEYSPACE IF NOT EXISTS %s with replication = {'class':'SimpleStrategy', 'replication_factor':1};", keyspaceName))
+	runCassandraQueryAndGetOutput(t, fmt.Sprintf("CREATE TABLE IF NOT EXISTS %s.%s(id timeuuid PRIMARY KEY, val text);", keyspaceName, tableName))
+}
+
+func LoadRowsInTable(t *testing.T, nbRows int, tableName, keyspaceName string) {
+	for i := 0; i < nbRows; i++ {
+		runCassandraQueryAndGetOutput(t, fmt.Sprintf("INSERT INTO %s.%s(id,val) values(now(), '%d');", keyspaceName, tableName, i))
+	}
+}
+
+// Medusa related functions
+func CreateMedusaSecretWithFile(t *testing.T, secretFile string) {
+	home, _ := os.UserHomeDir()
+	medusaSecretPath, _ := filepath.Abs(strings.Replace(secretFile, "~", home, 1))
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	k8s.KubectlApply(t, kubectlOptions, medusaSecretPath)
+}
+
+func PerformBackup(t *testing.T, backupName string) {
+	backupChartPath, err := filepath.Abs("../../charts/backup")
+	g(t).Expect(err).To(gomega.BeNil())
+
+	helmOptions := &helm.Options{
+		SetValues: map[string]string{
+			"name":                     backupName,
+			"cassandraDatacenter.name": "dc1",
+		},
+		KubectlOptions: k8s.NewKubectlOptions("", "", namespace),
+	}
+	// Start backup
+	helm.Install(t, helmOptions, backupChartPath, backupName)
+
+	// Wait for the backup to be finished
+	g(t).Eventually(func() int {
+		return backupIsFinished(t, backupName)
+	}, 2*time.Minute, 5*time.Second).Should(gomega.Equal(1))
+}
+
+func backupIsFinished(t *testing.T, backupName string) int {
+	var nodes []string
+	log.Printf("Checking if backup %s is finished...", backupName)
+	kubectlOptions := k8s.NewKubectlOptions("", "", namespace)
+	output, err := k8s.RunKubectlAndGetOutputE(t, kubectlOptions, "get", "cassandrabackup", backupName, "-o", "jsonpath='{.status.finished}'")
```

Comment:
This check might not always be right. `.status.finished` gets updated as each backup completes. The `.status.finishTime` is set once all of the backups are done.

Reply:
We're expecting a single backup here since the test runs with a single node.
I'll check `finishTime` though so we're sure to have the right behavior if we add nodes.

Reply:
Done 👍